dask/dataframe/io/demo.py

```diff
@@ -311,7 +309,7 @@ def make_dataframe_part(index_dtype, start, end, dtypes, columns, state_data, kw
     elif pd.api.types.is_integer_dtype(index_dtype):
         step = kwargs.get("freq")
         index = pd.RangeIndex(start=start, stop=end + step, step=step).astype(
-            index_dtype
+            index_dtype, copy=False
```

Comment:
Index.astype is always a shallow copy

Reply:
It's not quite the same though, right?

```
        copy : bool, default True
            By default, astype always returns a newly allocated object.
            If copy is set to False and internal requirements on dtype are
            satisfied, the original data is used to create a new Index
            or the original Index is returned.
```

Reply:
It's not exactly the same no, but performance-wise it is.

Index.copy() is a shallow copy by default while ``astype(..., copy=False)`` will return exactly the same object. The latter can create problems (it probably won't here), so I'd avoid it as much as possible. Sharing indexes between two different objects is generally not a great idea (and we will probably move away from that very soon in pandas)

Reply:
I see, that makes sense. The comment doesn't explain it as well. Thank you!

Reply:
It's still a shallow copy, which is very cheap. Does not matter in dask context

Reply:
Some more context if you Care :)

Indexes are immutable, so there is rarely a need for a deep copy since you can not change the data of an index anyway. There are some edge cases where you could, but we are removing that with Copy on Write. You can change some attributes of an Index though, like the name, hence why I'd recommend using different objects.